### ISSUE TITLE
Client logs out with "Stay signed in" enabled

### ISSUE THREAD

**Issue by c0d1ng** [shaarli/Shaarli#617]
Even if I check the _Stay signed_ in option, I still get logged out after an hour or so, which is quite annoying, since I need to log in again every time.
I am using PHP7 with shaarli v0.6.5 on debian.

**Comment by ArthurHoaro** [comment]
You guys are going to make me hate cookies, while I really like cookies. :(

  * What's your web server + version ?
  * Are you behind a reverse proxy ?
  * Can you check your cookie to see if it exists, and if it is attached to the right domain (Chromium debugger works better for cookies than Firefox).

![bigbank](https://cloud.githubusercontent.com/assets/1962678/17285500/608350aa-57c3-11e6-86ab-2e2ee6f86b67.png)

**Comment by c0d1ng** [comment]
My nginx version is 1.10.1, not using a reverse proxy.
The cookie does get created, also with a max-age of one year. But after a refresh of the page it gets deleted, and I need to login again.

**Comment by ArthurHoaro** [comment]
> But after a refresh of the page it gets deleted, and I need to login again.

Do you mean that you will have to login again in one hour or that you're immediately logged out?

The only time Shaarli touches this after the login, it's when the user is explicitly logged out (click on the Logout button, and a few other scenario). If the cookie is correctly set, I don't see any reason for it to disappear. Especially since cookies seems to work properly for the session.

However, I noticed that we don't use the same data to set the cookie between the "stay logged in" and the session. Can you try to replace, in `index.php`, around L400 (not sure in 0.6.5):
```php
setcookie('shaarli_staySignedIn', STAY_SIGNED_IN_TOKEN, time()+31536000, WEB_PATH);
```

by:
```php
$cookiedir = ''; 
if(dirname($_SERVER['SCRIPT_NAME'])!='/') { 
    $cookiedir = dirname($_SERVER["SCRIPT_NAME"]).'/';
}
setcookie('shaarli_staySignedIn', STAY_SIGNED_IN_TOKEN, time()+31536000, $cookiedir, $_SERVER['SERVER_NAME']);
```

**Comment by c0d1ng** [comment]
Thanks for your advise, however this didn't fix the issue for me unfortunately. :/

**Comment by ArthurHoaro** [comment]
Can you install a dummy instance, and share its credentials with us? Otherwise we won't be able to help any further without reproducing.

Also adding `var_dump($_SERVER);` at the begining of `index.php` could help.

**Comment by c0d1ng** [comment]
Ah, using vardump is indeed also a way to debug it, going to try that out :)

**Comment by ArthurHoaro** [comment]
Closing due to lack of feedback. Feel free to re-open this issue if you have additional information.

**Comment by c0d1ng** [comment]
Sorry for the long delay, but I just wanted to let you guys know that I guess I found the reason for this behaviour: Since in `index.php` line 181 the `STAY_SIGNED_IN_TOKEN` is also calculated by the current IP, every time my IP changed, I got logged out. This happened at least once a day by my ISP, or when using any VPN, or on my phone when switching from wifi to cullular etc..
What I did now is removing `$_SERVER['REMOTE_ADDR']` from the calculation. Didn't have any issues since.
